src/shared/components/data-table/components/table-controls/components/table-actions/components/download-data/DownloadData.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { ReactNode, useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom/client';
+import memoize from 'lodash/memoize';
+
+import useDataTable from 'src/shared/components/data-table/hooks/useDataTable';
+import { downloadTextAsFile } from 'src/shared/helpers/downloadAsFile';
+
+import { ControlledLoadingButton } from 'src/shared/components/loading-button';
+
+import { TableAction } from 'src/shared/components/data-table/dataTableTypes';
+import { LoadingState } from 'src/shared/types/loading-state';
+
+import styles from './DownloadData.scss';
+
+const getReactRenderer = memoize(() => {
+  const element = document.createElement('div');
+  const root = ReactDOM.createRoot(element);
+
+  return {
+    element,
+    renderer: root
+  };
+});
+
+const getReactNodeText = (node: ReactNode): string => {
+  const { element, renderer } = getReactRenderer();
+  renderer.render(node);
+  return element.innerText;
+};
+
+const DownloadData = () => {
+  const {
+    dispatch,
+    rows,
+    downloadFileName,
+    columns,
+    selectedAction,
+    hiddenRowIds,
+    downloadHandler
+  } = useDataTable();
+
+  const [downloadState, setDownloadState] = useState<LoadingState>(
+    LoadingState.NOT_REQUESTED
+  );
+  const allowComponentResetRef = useRef(true);
+
+  useEffect(() => {
+    allowComponentResetRef.current = true;
+    return () => {
+      allowComponentResetRef.current = false;
+    };
+  }, []);
+
+  const restoreDefaults = () => {
+    if (allowComponentResetRef.current) {
+      dispatch({
+        type: 'set_selected_action',
+        payload: TableAction.DEFAULT
+      });
+    }
+  };
+
+  const handleDownload = async () => {
+    setDownloadState(LoadingState.LOADING);
+    if (downloadHandler) {
+      try {
+        await downloadHandler();
+        setDownloadState(LoadingState.SUCCESS);
+        setTimeout(restoreDefaults, 1000);
+      } catch {
+        setDownloadState(LoadingState.ERROR);
+        setTimeout(() => setDownloadState(LoadingState.NOT_REQUESTED), 2000);
```

Comment:
You should probably also check for `allowComponentResetRef.current` in the callback to this setTimeout function. You are changing component's state here.

Reply:
That's done